apps/accounts/middleware.py

```diff
@@ -152,48 +139,35 @@ class HtkUserTokenAuthMiddleware(BaseHtkAuthMiddleware):
     def process_request(self, request):
         super().process_request(request)
 
-        if 'HTTP_AUTHORIZATION' in request.META:
+        token_type, token = self._parse_authorization_header(request)
+
+        if token_type == 'Bearer':
+            # The token is in the `Authorization` header
             self.is_explicit_auth = True
-            auth_header = request.META['HTTP_AUTHORIZATION']
-            parts = auth_header.split()
-            if len(parts) == 2:
-                token_type, token = parts
-                if token_type == 'Bearer':
-                    # Only Bearer token can continue
-                    pass
-                else:
-                    # unset token
-                    token = None
-
-                    if token_type == 'Basic':
-                        # Do not count Basic auth as explicit auth,
-                        # to give a chance for `HtkBasicAuthMiddleware` to work
-                        self.is_explicit_auth = False
-                    else:
-                        pass
-            else:
-                # unexpected authorization value
-                token = None
+            auth_user = authenticate(request=request, token=token)
+
         elif 'token' in request.GET:
             # implicit authenication token using URL params
+            self.is_explicit_auth = True
             token = request.GET['token']
+            auth_user = authenticate(request=request, token=token)
+
         else:
-            # No authentication token provided via available methods
-            token = None
+            # and give a chance for other auth middlewares to work
+            self.is_explicit_auth = False
+            auth_user = None
 
-        auth_user = (
-            authenticate(request=request, token=token) if token else None
-        )
         self._handle_auth_flow(request, auth_user)
 
 
 class HtkSocialAuthExceptionMiddleware(SocialAuthExceptionMiddleware):
     def get_redirect_uri(self, request, exception):
         """Redirect to LOGIN_ERROR_URL by default
-        Otherwise, go to the SOCIAL_AUTH_<STRATEGY>_LOGIN_ERROR_URL for that backend provider if specified
+        Otherwise, go to the SOCIAL_AUTH_<STRATEGY>_LOGIN_ERROR_URL for that backend
+        provider if specified
 
-        However, if user is logged in when the exception occurred, it is a connection failure
-        Therefore, always go to account settings page or equivalent
+        However, if user is logged in when the exception occurred, it is a connection
+        failure. Therefore, always go to account settings page or equivalent
```

Comment:
```suggestion
        However, if user is logged in when the exception occurred,
        it is a connection failure.
        Therefore, always go to account settings page or equivalent
```

Let's break the text differently. New sentence on new line...